src/resolvers/git.cr

```diff
@@ -242,8 +242,40 @@ module Shards
       false
     end
 
-    private def origin_changed?
-      (@origin_url ||= capture("git ls-remote --get-url origin").strip) != git_url
+    def origin_url
+      @origin_url ||= capture("git ls-remote --get-url origin").strip
+    end
+
+    # Returns whether origin URLs have differing hosts and/or paths.
+    protected def origin_changed?
+      return false if origin_url == git_url
+      return true if origin_url.nil? || git_url.nil?
+
+      origin_parsed = parse_git_uri(origin_url)
+      git_parsed = parse_git_uri(git_url)
+
+      (origin_parsed.host != git_parsed.host) || (origin_parsed.path != git_parsed.path)
+    end
+
+    # Parses a URI string, with additional support for ssh+git URI schemes.
+    private def parse_git_uri(raw_uri)
```

Comment:
```suggestion
    private def parse_uri(raw_uri)
```

The `git` naming feels redundant —we're already in the `GitResolver` class.

Reply:
👍